Backend/cabal.project

```diff
@@ -36,3 +36,12 @@ packages:
   test
   lib/producer
 
+-- 6 parallel jobs by default
+-- you can increase this number if on a more powerful CPU
+jobs: 6
```

Comment:
Does `cabal.project` not support [`$ncpus`](https://cabal.readthedocs.io/en/3.4/cabal-project.html#cfg-flag---jobs)?

Reply:
it does, but we also now use `-j4` in `ghc-options` of all packages, which allows them to use upto 4 cores to build.

using a number like 6 here means we can have 6 packages building in parallel, with each allowed to use upto 4 cores which means a maximum of 6 * 4 = 24 cores at a time, which is a lot more manageable.

if we allow `$ncpus` and then also allow each of those to use upto 4 cores each, that would saturate the system really quickly.

This was a number that should work for both laptops and desktops. if someone has a more powerful machine they can then explicitly bump this if required, or decrement it in case of an under-powered machine as mentioned in the relevant README sections.

Reply:
Okay, you want to state this (especially the ghc-options -j4 part) in the comment on lines 39,40.

Reply:
Added the commentary.